pages/product/[id].tsx

```diff
@@ -0,0 +1,68 @@
+import { NextPage, GetStaticProps, GetStaticPaths } from 'next';
+import { Heading, Divider, Alert, Spinner, Container } from '@theme-ui/components';
+import { Page } from 'components/layout';
+import { ProductCard } from 'components/products';
+import { takeshapeApiUrl, takeshapeApiKey } from 'lib/config';
+import { createApolloClient } from 'lib/apollo';
+import { Stripe_Product } from 'lib/takeshape/types';
+import {
+  GetStripeProduct,
+  GetStripeProductArgs,
+  GetStripeProductQuery,
+  GetStripeProducts,
+  StripeProducts
+} from 'lib/queries';
+
+interface ProductPageProps {
```

Comment:
Next has this util you can use:

```
import { InferGetStaticPropsType } from 'next'
```

and then, on your page types:

```
InferGetStaticPropsType<typeof getStaticProps>
```

Pretty neat! Maybe useful here?

Reply:
This is neat, I didn't know about this before!

It didn't end up being useful in this case, since I still need to provide a return type for `GetStaticProps` and these are so close together. I think this would be really useful if we were importing a `getStaticProps` function from a different module.